lib/navigation/crosscheck.cpp

```diff
@@ -0,0 +1,49 @@
+#include "crosscheck.hpp"
+
+namespace hyped::navigation{
+
+  Crosscheck::Crosscheck(){
+    //TODO: implement
+  }
+
+  bool Crosscheck::checkTrajectoryAgreement(const std::array<nav_t, kNumImus> imu_data,
```

Comment:
No I you're right for preprocessing! I need to make sure we deal with 3d imu entries. Will have to add functionality to get resultant and subtract calibration readings. Also will probably implement outlier detection on all 3 axis.

Reply:
But by the time we get to crosscheck, we'll be dealing with resultant accelerations. Will fix types tho. Since we will want to iterate over the sensors at some point, a struct is probably not the best solution?

Reply:
I don't have a definitive answer but here's how you may want to decide which one to go with for now (switching later down the line shouldn't be hard): Which type of access do you expect to see more? Iteration or point wise (e.g. imu_data.first or imu_data.at(0))? I would argue that the only thing you care about with the IMU array is that the order doesn't change; otherwise you are indifferent about which IMU goes where. So an array should be fine, as long as this guarantee is understood. 

One way you may want to fix this is by giving each `ImuData` an explicit ID (maybe talk to @SnickeyX to see whether there is an obvious choice, e.g. an address for this). This would allow you to verify that the data is indeed coming from the place that you thought.

Reply:
have changed to arrays in core types